server/Dockerfile

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright 2021 ABSA Group Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# Atum Service docker file
+
+# build via (docker root = project root):
+# docker build -t absaoss/atum-service:latest \
+# --build-arg BUILD_PROXY=http://my.cool.proxy.here:3128 \
+# --build-arg CONFIG=./path/to/my.awesome.local.application.properties \
+# --build-arg SSL=true \
+# --build-arg SSL_DNAME="CN=*.my.domain.com, OU=project1, O=mycorp, L=Johannesburg, ST=Gauteng, C=za" .
+# run via
+# docker run -p 8080:8080 -p 8443:8443 absaoss/atum-service:latest
+
+# test via:
+# http://localhost:8080/token/public-key
+# https://localhost:8443/token/public-key
+
+# Conditional Docker image creation idea: https://stackoverflow.com/a/60820156/1773349
+# change to OFF to disable SSL
+ARG SSL=true
+# Tomcat OS base image, feel free to use another (e.g. your hardened one)
+ARG BASE_IMAGE=tomcat:9-jdk8-corretto
+
+# --- Base image part (common for SSL true|false) ---
+FROM $BASE_IMAGE as base
+
+# Provide your proxy if needed, e.g. http://my.proxy.examle.com:3128
+ARG BUILD_PROXY
+# Override of the example application config is possible
+ARG CONFIG=./src/main/resources/application.properties
+# Provide path to the directory with LDAP certs in PEM format
+ARG LDAP_SSL_CERTS_PATH
+# ARG SSL_DNAME is defined below in the SSL-enabled image
+# In case you build the Dockerfile from another location than the default 'service' dir, provide a prefix to reach it
+ARG AS_PREFIX=.
+# Version of scala that server was built against
+ARG SCALA_VERSION=2.12
+
+LABEL org.opencontainers.image.authors="ABSA"
+
+# Copy Spring application properties
+COPY $CONFIG /opt/application.yaml
+
+# deploy as root application in tomcat
+COPY ${AS_PREFIX}/target/scala-${SCALA_VERSION}/*.war /usr/local/tomcat/webapps/ROOT.war
```

Comment:
this doesn't work, for some reason the actual path where this war is being put is `jvm-*/` and not `scala-*/` so it needs to be changed in `build.sbt` or here. I think that in LSD guys didn't use plugin `sbt-assembly` and their build looks simpler

```
 => [base 2/6] COPY ./src/main/resources/application.properties /opt/application.yaml                                                 0.0s
 => ERROR [base 3/6] COPY ./target/scala-2.12/*.war /usr/local/tomcat/webapps/ROOT.war                                                0.0s
------
 > [base 3/6] COPY ./target/scala-2.12/*.war /usr/local/tomcat/webapps/ROOT.war:
------
Dockerfile:59
--------------------
  57 |
  58 |     # deploy as root application in tomcat
  59 | >>> COPY ${AS_PREFIX}/target/scala-${SCALA_VERSION}/*.war /usr/local/tomcat/webapps/ROOT.war
  60 |
  61 |     ENV SPRING_CONFIG_LOCATION=/opt/application.yaml
--------------------
ERROR: failed to solve: lstat /var/lib/docker/tmp/buildkit-mount1632225145/target/scala-2.12: no such file or directory
```

Reply:
I have it working and tested, and `war` on my local machine is in the scala folder. 

![image](https://github.com/AbsaOSS/atum-service/assets/17467768/b5bb3e5b-9354-4df8-9529-8c75e6bd420c)

What command are you running?

Reply:
Pushed a fix, I had cached some stuff and I was still getting my war from `scala-2.12`.